packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -544,40 +496,84 @@ export function TransferPanel() {
     }
   }
 
-  const transfer = async () => {
-    const signerUndefinedError = 'Signer is undefined'
-
-    try {
-      setTransferring(true)
-      if (chainId !== networks.sourceChain.id) {
-        await switchNetworkAsync?.(networks.sourceChain.id)
-      }
-    } finally {
-      setTransferring(false)
+  async function hasDestinationAddressError() {
+    const destinationAddressError = await getDestinationAddressError({
+      destinationAddress,
+      isSmartContractWallet,
+      isTeleportMode
+    })
+    if (destinationAddressError) {
+      console.error(destinationAddressError)
+      return true
     }
+    return false
+  }
 
+  async function isTransferAllowed(isCctp: boolean) {
```

Comment:
Could just get `isCctp` directly instead of passing the prop

Reply:
Also this method uses a lot of hook values directly, are we sure we are not running into stale values or race conditions here?